src/GitHub.VisualStudio/Services/UIProvider.cs

```diff
@@ -85,7 +80,7 @@ public object TryGetService(Type serviceType)
                 initializingLogging = true;
                 try
                 {
-                    var logging = TryGetService<ILoggingConfiguration>();
+                    var logging = TryGetService(typeof(ILoggingConfiguration)) as ILoggingConfiguration;
```

Comment:
It wasn't obvious to me why you're making this change and not just calling the cleaner looking generic method. Then I read your PR comment.

> UIProvider should only use non-generic variants of GetService/TryGetService so that calls don't spill over to ServiceProviderExtensions.cs and potentially cause recursive calls.

That's a bit tricky and nuanced and people looking at this code in the future might be really tempted to call that generic method. One idea is to add a private generic `TryGetService` to this class that then calls the non-generic one for you and document that one method with the comment you made. That way nobody will be tempted to change this call in a breaking manner.

Thoughts?